src/contrib/cluster/Akka.Cluster.Sharding/DDataShardCoordinator.cs

```diff
@@ -7,356 +7,662 @@
+                    case UpdateSuccess m when m.Key.Equals(_coordinatorStateKey) && m.Request.Equals(evt):
+                        if (!waitingForRememberShard)
+                        {
+                            Log.Debug("{0}: The coordinator state was successfully updated with {1}", TypeName, evt);
+                            if (shardId != null)
+                                Timers.Cancel(RememberEntitiesTimeoutKey);
+                            UnbecomeAfterUpdate(evt, afterUpdateCallback);
+                        }
                         else
-                            Context.Become(WaitingForUpdate(e, afterUpdateCallback, newRemainingKeys));
+                        {
+                            Log.Debug("{0}: The coordinator state was successfully updated with {1}, waiting for remember shard update",
+                                TypeName,
+                                evt);
+                            Context.Become(
+                                WaitingForUpdate(
+                                    evt,
+                                    shardId,
+                                    waitingForStateWrite = false,
+                                    waitingForRememberShard = true,
+                                    afterUpdateCallback: afterUpdateCallback));
+                        }
                         return true;
 
-                    case UpdateTimeout timeout when timeout.Key.Equals(_coordinatorStateKey) && timeout.Request.Equals(e):
-                        Log.Error("The ShardCoordinator was unable to update a distributed state within 'updating-state-timeout': {0} ({1}), event={2}", _writeConsistency.Timeout, _terminating ? "terminating" : "retrying", e);
-
+                    case UpdateTimeout m when m.Key.Equals(_coordinatorStateKey) && m.Request.Equals(evt):
+                        Log.Error(
+                            "{0}: The ShardCoordinator was unable to update a distributed state within 'updating-state-timeout': {1} millis ({2}). " +
+                            "Perhaps the ShardRegion has not started on all active nodes yet? event={3}",
+                            TypeName,
+                            _stateWriteConsistency.Timeout.TotalMilliseconds,
+                            _terminating ? "terminating" : "retrying",
+                            evt);
                         if (_terminating)
                         {
                             Context.Stop(Self);
                         }
                         else
                         {
                             // repeat until UpdateSuccess
-                            SendCoordinatorStateUpdate(e);
+                            SendCoordinatorStateUpdate(evt);
                         }
-
                         return true;
 
-                    case UpdateSuccess success when success.Key.Equals(_allShardsKey) && success.Request is string newShard:
```

Comment:
So we don't handle this case now - what occurs when we receive an `UpdateSuccess` when the key is not equal to `_coordinatorStateKey`? Just an `Unhandled` message?